sktime/forecasting/squaring_residuals.py

```diff
@@ -0,0 +1,399 @@
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements the probabilistic Squaring Residuals forecaster."""
+
+__all__ = ["SquaringResiduals"]
+__author__ = ["kcc-lion"]
+
+from warnings import warn
+
+import pandas as pd
+
+from sktime.datatypes._convert import convert_to
+from sktime.forecasting.base import BaseForecaster
+from sktime.forecasting.model_selection import ExpandingWindowSplitter
+from sktime.forecasting.naive import NaiveForecaster
+
+
+class SquaringResiduals(BaseForecaster):
+    r"""Compute the prediction variance based on a separate forecaster.
+
+    Wraps a `forecaster` with another `variance_forecaster` object that
+    allows for quantile and interval estimation by fitting the
+    `variance_forecaster` to the rolling residuals.
+
+    Fitting proceeds as follows:
+    Let :math:`t_1, \dots, t_N` be the train set.
+    Let `steps_ahead` be a positive integer indicating the steps ahead
+    we want to forecast the residuals. Let `initial_window` be
+    the minimal number of observations to which the forecaster is fitted.
+
+    1. For :math:`i = initial\_window, \dots, N - steps\_ahead`
+        a. Train/Update forecaster A on :math:`y(t_1), \dots, y(t_i)`
+        b. Make point prediction for :math:`t_{i+steps\_ahead}` to get
+           :math:`\hat{y}(t_{i+steps\_ahead})`
+        c. Compute the residual for :math:`t_{i+steps\_ahead}` with
+           :math:`r(t_{i+steps\_ahead}) := y(t_{i+steps\_ahead})
+           - \hat{y}(t_{i+steps\_ahead})`
+        d. Apply  :math:`e(t_{i+steps\_ahead}) := h(r(t_{i+steps\_ahead}))`
+           where :math:`h(x)` is given by :math:`strategy`
+    2. Train `variance_forecaster` on
+       :math:`e(t_{initial\_window+steps\_ahead}), \dots, e(t_{N})`
+
+    Prediction for :math:`t_{N+steps\_ahead}` is done as follows:
+
+    1. Use `forecaster` to predict location param :math:`\hat{y}(t_{N+steps\_ahead})`
+    2. Use `variance_forecaster` to predict scale param :math:`e(t_{N+steps\_ahead})`
+    3. Calculate prediction intervals based on e.g. normal assumption
+       :math:`N(\hat{y}(t_{N+steps\_ahead}),  e(t_{N+steps\_ahead}))`
+
+    Parameters
+    ----------
+    forecaster : sktime.estimator, optional
```

Comment:
`sktime.estimator` is incorrect (there is no such module)

suggest:
`sktime` forecaster, `BaseForecaster` descendant